src/util/Sweepers.js

```diff
@@ -0,0 +1,450 @@
+'use strict';
+
+const { Events, ThreadChannelTypes, SweeperKeys } = require('./Constants');
+const { TypeError } = require('../errors/DJSError.js');
+
+/**
+ * @typedef {Function} GlobalSweepFilter
+ * @returns {Function|null} Return `null` to skip sweeping, otherwise a function passed to `sweep()`,
+ * See {@link [Collection#sweep](https://discord.js.org/#/docs/collection/main/class/Collection?scrollTo=sweep)}
+ * for the definition of this function.
+ */
+
+/**
+ * A container for all cache sweeping intervals and their associated sweep methods.
+ */
+class Sweepers {
+  constructor(client, options) {
+    /**
+     * The client that instantiated this
+     * @type {Client}
+     * @readonly
+     */
+    Object.defineProperty(this, 'client', { value: client });
+
+    /**
+     * The options the sweepers were instantiated with
+     * @type {SweeperOptions}
+     */
+    this.options = options;
+
+    /**
+     * A record of interval timeout that is used to sweep the indicated items, or null if not being swept
+     * @type {Object<SweeperKey, ?Timeout>}
+     */
+    this.intervals = {};
```

Comment:
We can probably define all the possible keys here, so we can see the potential list of possible sweepers in this place. Also to avoid [shape changing](https://mathiasbynens.be/notes/shapes-ics).

Reply:
I can use Object.fromEntries to avoid shape changing I believe. The list already exists in Constants and is just a ctrl-click away (will be right in that call now. If you still want to manually call it out, that can be done though.